client/identities.go

```diff
@@ -47,6 +48,12 @@ type LocalIdentity struct {
 	UserID *uint32 `json:"user-id" yaml:"user-id"`
 }
 
+// BasicIdentity holds identity configuration specific to the "basic" type
+// (for HTTP basic authentication).
+type BasicIdentity struct {
+	Password string `json:"password" yaml:"password"`
```

Comment:
This may be overkill, but is this password hashed and salted? If so, I think it may be worth noting this in a doc comment (like in the `BasicIdentity` in internals). Especially as this is a public field on a public type, there's a small but horrifying possibility that this secret could end up in logs or plaintext on disk

I also notice that the internals `BasicIdentity` has the same layout as `BasicIdentity`. Is this intentional and something which needs to be maintained? If so, is it worth adding a comments on each to state that if one is changed, the other should be too? It may also be possible to assert that they have the same layout at compile time

Reply:
Yes, it's hashed and salted. A comment is added elsewhere but worth mentioning here, thanks for pointing it out!

As for the internal/duplication of the same struct: Yes, it's intentional, because the client has a separate definition of it and should not depend on the internal one.